### DISCUSSION TITLE
Error in Compiling LIS

### DISCUSSION THREAD

**Discussion by Lise-Li** [NASA-LIS/LISF#1421] [Q&A] [answered]
Hello, I'm compiling LIS-7.4.0, I have installed ESMF-8.4.2, openjpeg-2.4.0, eccodes-2.23.0 and loaded openmpi/2.1.5 and GNU Compiler11.2. This is my environment settings:
export LIS_ARCH=linux_gfortran
export LIS_FC=mpif90
export LIS_CC=mpicc
export LIS_MODESMF=/home/name/project/esmf/esmf_install/mod/modO/Linux.gfortran.64.openmpi.default
export LIS_LIBESMF=/home/name/project/esmf/esmf_install/lib/libO/Linux.gfortran.64.openmpi.default
export LIS_OPENJPEG=/home/name/project/openjpeg-2.4.0
export LIS_ECCODES=/home/name/project/eccodes
export LIS_NETCDF=/home/name/project/tools/netcdf
export LIS_HDF5=/home/name/project/tools/hdf5
However, the following error is reported after running the compile:
mpif90 -c -ffree-line-length-0 -O2  -fbacktrace -DSPMD -fconvert=big-endian -DHIDE_SHR_MSG -DNO_SHR_VMATH -DGFORTRAN -DLINUX  -I/home/lihuijie/project/esmf/esmf_install/mod/modO/Linux.gfortran.64.openmpi.default -I/home/lihuijie/project/eccodes/include/ -I/home/lihuijie/project/tools/netcdf/include/ -I/home/lihuijie/project/tools/hdf5/include/ -DLIS_JULES -I./ -I../core/ -I../params/gfrac/SPORTDaily/ -I../params/gfrac/VIIRSDaily/ -I../surfacemodels/land/vic.4.1.1/ -I../surfacemodels/land/vic.4.1.1/physics/ -I../surfacemodels/land/vic.4.1.2.l/ -I../surfacemodels/land/vic.4.1.2.l/physics/ -I../surfacemodels/land/awral.6.0.0/physics/ -I../surfacemodels/land/rdhm.3.5.6/ -I../surfacemodels/land/rdhm.3.5.6/sac/ -I../surfacemodels/land/rdhm.3.5.6/frz/ -I../surfacemodels/land/rdhm.3.5.6/snow17/ -I../surfacemodels/land/rdhm.3.5.6/sachtet/ ../core/LIS_constantsMod.F90
mpif90 -c -ffree-line-length-0 -O2  -fbacktrace -DSPMD -fconvert=big-endian -DHIDE_SHR_MSG -DNO_SHR_VMATH -DGFORTRAN -DLINUX  -I/home/lihuijie/project/esmf/esmf_install/mod/modO/Linux.gfortran.64.openmpi.default -I/home/lihuijie/project/eccodes/include/ -I/home/lihuijie/project/tools/netcdf/include/ -I/home/lihuijie/project/tools/hdf5/include/ -DLIS_JULES -I./ -I../core/ -I../params/gfrac/SPORTDaily/ -I../params/gfrac/VIIRSDaily/ -I../surfacemodels/land/vic.4.1.1/ -I../surfacemodels/land/vic.4.1.1/physics/ -I../surfacemodels/land/vic.4.1.2.l/ -I../surfacemodels/land/vic.4.1.2.l/physics/ -I../surfacemodels/land/awral.6.0.0/physics/ -I../surfacemodels/land/rdhm.3.5.6/ -I../surfacemodels/land/rdhm.3.5.6/sac/ -I../surfacemodels/land/rdhm.3.5.6/frz/ -I../surfacemodels/land/rdhm.3.5.6/snow17/ -I../surfacemodels/land/rdhm.3.5.6/sachtet/ ../interp/easeV2_utils.F90
mpif90 -c -ffree-line-length-0 -O2  -fbacktrace -DSPMD -fconvert=big-endian -DHIDE_SHR_MSG -DNO_SHR_VMATH -DGFORTRAN -DLINUX  -I/home/lihuijie/project/esmf/esmf_install/mod/modO/Linux.gfortran.64.openmpi.default -I/home/lihuijie/project/eccodes/include/ -I/home/lihuijie/project/tools/netcdf/include/ -I/home/lihuijie/project/tools/hdf5/include/ -DLIS_JULES -I./ -I../core/ -I../params/gfrac/SPORTDaily/ -I../params/gfrac/VIIRSDaily/ -I../surfacemodels/land/vic.4.1.1/ -I../surfacemodels/land/vic.4.1.1/physics/ -I../surfacemodels/land/vic.4.1.2.l/ -I../surfacemodels/land/vic.4.1.2.l/physics/ -I../surfacemodels/land/awral.6.0.0/physics/ -I../surfacemodels/land/rdhm.3.5.6/ -I../surfacemodels/land/rdhm.3.5.6/sac/ -I../surfacemodels/land/rdhm.3.5.6/frz/ -I../surfacemodels/land/rdhm.3.5.6/snow17/ -I../surfacemodels/land/rdhm.3.5.6/sachtet/ ../interp/map_utils.F90
mpif90 -c -ffree-line-length-0 -O2  -fbacktrace -DSPMD -fconvert=big-endian -DHIDE_SHR_MSG -DNO_SHR_VMATH -DGFORTRAN -DLINUX  -I/home/lihuijie/project/esmf/esmf_install/mod/modO/Linux.gfortran.64.openmpi.default -I/home/lihuijie/project/eccodes/include/ -I/home/lihuijie/project/tools/netcdf/include/ -I/home/lihuijie/project/tools/hdf5/include/ -DLIS_JULES -I./ -I../core/ -I../params/gfrac/SPORTDaily/ -I../params/gfrac/VIIRSDaily/ -I../surfacemodels/land/vic.4.1.1/ -I../surfacemodels/land/vic.4.1.1/physics/ -I../surfacemodels/land/vic.4.1.2.l/ -I../surfacemodels/land/vic.4.1.2.l/physics/ -I../surfacemodels/land/awral.6.0.0/physics/ -I../surfacemodels/land/rdhm.3.5.6/ -I../surfacemodels/land/rdhm.3.5.6/sac/ -I../surfacemodels/land/rdhm.3.5.6/frz/ -I../surfacemodels/land/rdhm.3.5.6/snow17/ -I../surfacemodels/land/rdhm.3.5.6/sachtet/ ../core/LIS_mpiMod.F90
../core/LIS_mpiMod.F90:24:7:

   24 |   use mpi
      |       1
Fatal Error: Cannot read module file ‘/software/openmpi/2.1.5/lib/mpi.mod’ opened at (1), because it was created by a different version of GNU Fortran
compilation terminated.
make: *** [Makefile:140: LIS_mpiMod.o] Error 1

[ERR] Compile failed

**Comment by Lise-Li** [comment]
![configure](https://github.com/NASA-LIS/LISF/assets/145978291/7cbf4f49-07e7-4aca-8951-747514a0be9f)
this is my configure.lis

**Comment by MariRiba** [comment]
Hi
I guess the minimum openmpi version is Open MPI version 4 as given in [https://github.com/NASA-LIS/LISF/blob/master/docs/LISF_installation_guide/dependencies.adoc](url) .

**Comment by jvgeiger** [comment, marked as answer]
Hello,

I have not tried compiling LIS with OpenMPI 2.  As the documentation says, I used OpenMPI version 4.

But the actual problem is 

```
Fatal Error: Cannot read module file ‘/software/openmpi/2.1.5/lib/mpi.mod’ opened at (1), because it was created by a different version of GNU Fortran
```
This says that OpenMPI 2 was compiled with a different version the GNU Fortran compiler than you are using to compile LIS.  This is a limitation of Fortran module files. In general, everything that contains a Fortran module (which compiles to a *.mod file) must be compiled with the same Fortran compiler.

You need an MPI library compiled with GNU compilers version 11.2.

**Reply by Lise-Li** [reply]
Hi, @jvgeiger ! Thank you for your reply! The problems have been solved! Thank you!
